modules/atomic-widgets/base/atomic-widget-base.php

```diff
@@ -49,4 +55,37 @@ final public function get_stack( $with_common_controls = true ) {
 			'tabs' => [ 'content' => 'content' ],
 		];
 	}
+
+	final public function get_atomic_settings(): array {
+		$schema = static::get_props_schema();
+		$raw_settings = $this->get_settings();
+
+		$transformed_settings = [];
+
+		foreach ( $schema as $key => $prop ) {
+			if ( array_key_exists( $key, $raw_settings ) ) {
+				$transformed_settings[ $key ] = $raw_settings[ $key ];
+			} else {
+				$transformed_settings[ $key ] = $prop->get_default();
+			}
+		}
+
```

Comment:
```suggestion
    return new Collection( $schema )->pluck( 'default' )->merge( $raw_settings )->to_array();
```

Reply:
Do you want Rotem to kill me?

Reply:
you can decode it first....
But just a recommendation